bin/warnhist

```diff
@@ -294,19 +503,13 @@ while (<>) {
 
     # Check for extra ':' followed by more text in original warning string,
     #  and append the ':' and text back onto the parsed warning
-    #  (Use 'length $extra' idiom to avoid warning when $extra is undefined)
-    if(length $extra ) {
+    if(defined $extra) {
         $warning = join ':', $warning, $extra;
     }
-    if(length $extra2 ) {
+    if(defined $extra2) {
         $warning = join ':', $warning, $extra2;
     }
 
-    # Restore the C++ '::' symbol now that we've parsed out the parts of the line
```

Comment:
Moved down, to avoid 'undefined variable' warning from Perl about $warning variable.